### PR TITLE
[14.0][FIX] contract_payment_mode: set default payment mode by contract type

### LINKED ISSUE #923: [14.0][15.0] Incorrect selection of Payment Mode from the Partner data
The automatic selection of the Payment Mode when the Partner is selected in the Supplier Contract is wrong Payment Mode (inbound), the correct one would be outbound Payment Mode.

## Module
[contract_payment_mode](https://github.com/OCA/contract/tree/14.0/contract_payment_mode)

## To Reproduce
**Affected versions**:

1. Create a Supplier Contract
2. Select the Supplier "Deco"
3. The Payment Mode broght into the contract will be an **Inbound** Payment Mode selected on the partner data.

14.0 version
image
15.0 version:
image
16.0 version:
There isn't the contract app in this version yet.

Expected information on the partner data
image

Demonstration Video:
https://user-images.githubusercontent.com/20867090/216049926-86119c73-684f-47ab-9ae5-bc8cea3a60d7.mp4

**Expected behavior**
The Outbound Payment Mode seleted on the partner must be brought into the supplier contract

@marcelsavegnago @augustodinizl

### PR DESCRIPTION
**Pull request by marcelsavegnago**
ref #923 

cc @douglascstd @pedrobaeza

### CODE REVIEW

File: contract_payment_mode/models/contract.py [hunk 1]
```diff
@@ -13,7 +13,10 @@ class ContractContract(models.Model):
 
     @api.onchange("partner_id")
     def on_change_partner_id(self):
-        self.payment_mode_id = self.partner_id.customer_payment_mode_id.id
+        if self.contract_type == "purchase":
+            self.payment_mode_id = self.partner_id.supplier_payment_mode_id.id
```
Comment (pedrobaeza) [inline contract_payment_mode/models/contract.py]: Now that you are here, let's improve this for being aware of the company per document feature:

```suggestion
            self.payment_mode_id = self.with_company(self.company_id).partner_id.supplier_payment_mode_id.id
```

Comment (marcelsavegnago) [reply contract_payment_mode/models/contract.py]: @pedrobaeza Done

File: contract_payment_mode/models/contract.py [hunk 2]
```diff
@@ -13,7 +13,14 @@ class ContractContract(models.Model):
...
     def on_change_partner_id(self):
-        self.payment_mode_id = self.partner_id.customer_payment_mode_id.id
+        if self.contract_type == "purchase":
+            self.payment_mode_id = self.with_company(
+                self.company_id
+            ).partner_id.supplier_payment_mode_id.id
+        else:
+            self.payment_mode_id = self.with_company(
+                self.company_id
+            ).partner_id.customer_payment_mode_id.id
```
Comment (pedrobaeza) [inline contract_payment_mode/models/contract.py]: More readable, but a nitpicking:
```suggestion
        partner = self.with_company(self.company_id).partner_id
        if self.contract_type == "purchase":
            self.payment_mode_id = partner.supplier_payment_mode_id.id
        else:
            self.payment_mode_id = partner.customer_payment_mode_id.id
```

**Review by pedrobaeza — APPROVED** [review]

File: contract_payment_mode/models/contract.py (hunk 2 above)
Comment (marcelsavegnago) [reply contract_payment_mode/models/contract.py]: @pedrobaeza perfect.. I'll adjust.
Thanks

Comment (marcelsavegnago) [reply contract_payment_mode/models/contract.py]: @pedrobaeza Done.

**Review by douglascstd — APPROVED** [review]
[FUNCTIONAL TESTE][APPROVED]
Supplier contract:
image

Customer contract:
image

Partner information:
image

**Pull request merged**